explorer/client/src/components/search/Search.tsx

```diff
@@ -1,85 +1,183 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import React, { useState, useCallback, useContext } from 'react';
+import React, {
+    useState,
+    useCallback,
+    useContext,
+    useRef,
+    useEffect,
+} from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { ReactComponent as SearchIcon } from '../../assets/search.svg';
 import { NetworkContext } from '../../context';
-import { navigateWithUnknown } from '../../utils/searchUtil';
+import {
+    navigateWithCategory,
+    SEARCH_CATEGORIES,
+} from '../../utils/searchUtil';
+import { type ResultType } from './SearchResultType';
+import SearchResults from './SearchResults';
 
 import styles from './Search.module.css';
 
 function Search() {
-    const [input, setInput] = useState('');
-    const [network] = useContext(NetworkContext);
     const navigate = useNavigate();
+    const wrapperref = useRef<HTMLDivElement>(null);
+    const [network] = useContext(NetworkContext);
+    const [input, setInput] = useState('');
+
+    const [result, setResult] = useState<ResultType[] | null>(null);
+    const [resultIndex, setResultIndex] = useState(0);
+
+    const handleClickOutside = (event: MouseEvent): void => {
+        if (
+            wrapperref.current &&
+            !wrapperref.current.contains(event.target as Node)
+        ) {
+            setResult(null);
+            setInput('');
+        }
+    };
+
+    const handleKeyPress = useCallback(
+        (event: KeyboardEvent): void => {
+            // Press Down Key or Tab
+            if (event.keyCode === 40 || event.keyCode === 9) {
+                event.preventDefault();
+                setResultIndex((prevIndex) =>
+                    result?.length && prevIndex < result.length - 1
+                        ? prevIndex + 1
+                        : 0
+                );
+            }
+
+            // Press Up Key
+            if (event.keyCode === 38) {
+                event.preventDefault();
+                setResultIndex((prevIndex) =>
+                    prevIndex > 0
+                        ? prevIndex - 1
+                        : result?.length && result.length > 1
+                        ? result.length - 1
+                        : prevIndex
+                );
+            }
+        },
+        [result]
+    );
+
+    // Clicking Outside the Search Bar and Results should clear the search
```

Comment:
This UX was a little weird for me, that I could paste a valid address, click outside, and have it cleared. cc @mystie711

Reply:
Thanks for pointing out this detail. Hadn't thought this far earlier.

There are a couple options,

1) we keep the url in the address bar until it is manually cleared
2) we do clear out the address like it's implemented today. We implement a "Recent search" categories with the last three searches in the search suggestions UI. When the user starts typing then the recent searches disappears. Ofcourse, to complete this experience we will need to provide a "Clear searches" action at the other end of the "Recent" row.

Ideally, I would love to have 2 but totally understand if this gets a lot bigger chunk of work. In that case, let's go with 1.

@Andrew47 @Jordan-Mysten

Reply:
For 2, I’m curious how recent searches would denote the category?

Reply:
Recent searches are meant to be category agnostic. Just preserving search keywords for easy access.

Captured here,
https://www.figma.com/file/kSD0qXCIDZA4KQVNepA5hR/02-Explorer?node-id=67%3A18134

Reply:
That makes sense to me. I'm happy with either of those approaches, if we do 1 we can open an issue to eventually implement 2 as well.

Reply:
This makes sense - I'll implement (1) for now and open an issue for (2).

Reply:
(1) has been implemented